proto/supply/v1beta/genesis.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+package chainmain.supply.v1beta1;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/crypto-com/chain-main/x/supply/types";
+
+// VestingAccounts stored in keeper
+message VestingAccounts {
+    // addresses defines addresses of all the vesting accounts at genesis
+    repeated string addresses = 1;
```

Comment:
does this need a proto definition / duplicate information to what's in genesis? why not just iterate accounts in genesis and check the account type?

Reply:
I haven't seen any examples of directly reading whole genesis file in a modules `InitChain`. Instead the common pattern in `cosmos-sdk` is to have separate `app_state` for each module in `genesis.json` for initialisation. I've added code to inject vesting accounts in `genesis.json` in `add-genesis-account` command (similar to what we do for `bank` and `auth` modules).

Reply:
how about getting vesting information through keeper?
not directly refering to original genesis.json(of tendermint)

Reply:
@leejw51crypto I don't think we can do that. For that, we'll have to loop through all the accounts in the system.

Reply:
how about in InitGenesis
use this api of AccountKeeper
iterate all accounts , check vesting, append ?

// GetAllAccounts returns all accounts in the accountKeeper.
func (ak AccountKeeper) GetAllAccounts(ctx sdk.Context) (accounts []types.AccountI) {
	ak.IterateAccounts(ctx, func(acc types.AccountI) (stop bool) {
		accounts = append(accounts, acc)
		return false
	})

	return accounts
}

Reply:
Won't this iterate over all the accounts? We only want to iterate over accounts which were specified in `genesis.json`.

Reply:
yes, but in genesis init part,
accounts are a few numbers, so i think it will be very fast

Reply:
genesis.json can be large-ish if the state is exported from a previous network -- e.g. in cosmohub, it's several MBs -- but since "initgenesis" is only executed once, it's probably ok to iterate over the accounts.

So I suggest either:
1. drop the duplicate data and just compute it once from genesis
2. or add extra validations (https://github.com/crypto-com/chain-main/pull/258/files#diff-f995c0703ad210807c8ed69e34026fbcaa1cfe5a0acb4d9f28cafdb9c6d98254R23) to make sure there's no mismatch between genesis accounts and this duplicate supply accounts.

maybe 1. is simpler?

Reply:
option 1 seems good

Reply:
Done.